client/tests/input_selection/timelock.rs

```diff
@@ -0,0 +1,184 @@
+
+use iota_client::{api::input_selection::InputSelection, block::protocol::protocol_parameters, Error};
+
+use crate::input_selection::{
+    addresses, build_inputs, build_outputs, unsorted_eq, Build::Basic, BECH32_ADDRESS_ED25519_0,
+    BECH32_ADDRESS_ED25519_1,
+};
+
+#[test]
+fn one_output_timelock_not_expired() {
+    let protocol_parameters = protocol_parameters();
+
+    let inputs = build_inputs(vec![Basic(
+        2_000_000,
+        BECH32_ADDRESS_ED25519_0,
+        None,
+        None,
+        None,
+        Some(200),
+        None,
+    )]);
+    let outputs = build_outputs(vec![Basic(
+        1_000_000,
+        BECH32_ADDRESS_ED25519_1,
+        None,
+        None,
+        None,
+        None,
+        None,
+    )]);
+
+    let selected = InputSelection::new(
+        inputs,
+        outputs,
+        addresses(vec![BECH32_ADDRESS_ED25519_0]),
+        protocol_parameters,
+    )
+    .timestamp(100)
+    .select();
+
+    assert!(matches!(selected, Err(Error::NoAvailableInputsProvided)));
+}
+
+#[test]
+fn timelock_equal_timestamp() {
+    let protocol_parameters = protocol_parameters();
+
+    let inputs = build_inputs(vec![Basic(
+        2_000_000,
+        BECH32_ADDRESS_ED25519_0,
+        None,
+        None,
+        None,
+        Some(200),
+        None,
+    )]);
+    let outputs = build_outputs(vec![Basic(
+        2_000_000,
+        BECH32_ADDRESS_ED25519_1,
+        None,
+        None,
+        None,
+        None,
+        None,
+    )]);
+
+    let selected = InputSelection::new(
+        inputs.clone(),
+        outputs.clone(),
+        addresses(vec![BECH32_ADDRESS_ED25519_0]),
+        protocol_parameters,
+    )
+    .timestamp(200)
+    .select()
+    .unwrap();
+
+    assert!(unsorted_eq(&selected.inputs, &inputs));
+    assert!(unsorted_eq(&selected.outputs, &outputs));
+}
+
+#[test]
+fn two_outputs_one_timelock_expired() {
+    let protocol_parameters = protocol_parameters();
+
+    let inputs = build_inputs(vec![
+        Basic(2_000_000, BECH32_ADDRESS_ED25519_0, None, None, None, Some(200), None),
+        Basic(2_000_000, BECH32_ADDRESS_ED25519_0, None, None, None, Some(50), None),
+    ]);
+    let outputs = build_outputs(vec![Basic(
+        2_000_000,
+        BECH32_ADDRESS_ED25519_1,
+        None,
+        None,
+        None,
+        None,
+        None,
+    )]);
+
+    let selected = InputSelection::new(
+        inputs.clone(),
+        outputs.clone(),
+        addresses(vec![BECH32_ADDRESS_ED25519_0]),
+        protocol_parameters,
+    )
+    .timestamp(100)
+    .select()
+    .unwrap();
+
+    assert_eq!(selected.inputs.len(), 1);
+    assert_eq!(selected.inputs[0], inputs[1]);
+    assert!(unsorted_eq(&selected.outputs, &outputs));
+}
+
+#[test]
+fn two_outputs_one_timelock_expired_2() {
+    let protocol_parameters = protocol_parameters();
+
+    let inputs = build_inputs(vec![
+        Basic(2_000_000, BECH32_ADDRESS_ED25519_0, None, None, None, Some(200), None),
```

Comment:
It's not expired if the timestamp is 100?

Reply:
Maybe the name is not very clear. It's a copy paste of the other test where the output was expired, but this one doesn't even have an expiration. The test was not aimed at the 200 expiration but at the missing one.

Reply:
I can rename

Reply:
Done